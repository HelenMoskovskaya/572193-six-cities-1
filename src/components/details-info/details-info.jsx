import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Operation} from '../../reducer/data/data.js';
import Bookmark from '../bookmark/bookmark.jsx';
import {propTypesConstans} from '../../prop-types.js';
import {CountRating} from '../../constans.js';

const DetailsInfo = (props) => {
  const {offer, changeFavorites} = props;
  return <React.Fragment>
    {offer.isPremium && <div className="property__mark">
      <span>Premium</span>
    </div>}
    <div className="property__name-wrapper">
      <h1 className="property__name">
        {offer.title}
      </h1>
      <Bookmark
        isFavorite={offer.isFavorite}
        className={`property__bookmark-button`}
        changeFavorites={()=> {
          changeFavorites(offer);
        }}/>
    </div>
    <div className="property__rating rating">
      <div className="property__stars rating__stars">
        <span style={{width: `${Math.round(offer.rating) / CountRating.MAX_POINT * CountRating.CONVERTER_FOR_PERCENT}` + `%`}}></span>
        <span className="visually-hidden">Rating</span>
      </div>
      <span className="property__rating-value rating__value">{offer.rating}</span>
    </div>
    <ul className="property__features">
      <li className="property__feature property__feature--entire">
        {offer.type}
      </li>
      <li className="property__feature property__feature--bedrooms">
        {offer.bedrooms}
      </li>
      <li className="property__feature property__feature--adults">
        {offer.maxAdults}
      </li>
    </ul>
    <div className="property__price">
      <b className="property__price-value">&euro;{offer.price}</b>
      <span className="property__price-text">&nbsp;night</span>
    </div>
    <div className="property__inside">
      <h2 className="property__inside-title">What&apos;s inside</h2>
      <ul className="property__inside-list">
        {offer.goods.map((good) => {
          return <li key={good} className="property__inside-item">{good}</li>;
        })}
      </ul>
    </div>
  </React.Fragment>;
};

const mapDispatchToProps = (dispatch) => ({
  changeFavorites: (offer) => {
    dispatch(Operation.changeFavorites(offer));
  }
});

DetailsInfo.propTypes = {
  offer: propTypesConstans.OFFER,
  changeFavorites: PropTypes.func.isRequired,
};

export {DetailsInfo};
export default connect(null, mapDispatchToProps)(DetailsInfo);
