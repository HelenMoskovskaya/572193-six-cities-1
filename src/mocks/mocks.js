export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 3
  }
];

export const mockOffer = {
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
  images: [
    `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
  ],
  title: `Waterfront with extraordinary view`,
  isFavorite: false,
  isPremium: true,
  rating: 2.7,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 8,
  price: 393,
  goods: [
    `Breakfast`,
    `Washer`,
    `Laptop friendly workspace`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  location: {
    latitude: 51.214402,
    longitude: 6.764314000000001,
    zoom: 16
  },
  id: 1
};

export const cityMock = {
  name: `Amsterdam`,
  location: {
    latitude: 51.225402,
    longitude: 6.776314,
    zoom: 13
  }
};

export const cities = [`Amsterdam`, `Paris`, `Colone`];

export const userData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`,
  isPro: false
};

export const mockFavorites = {
  "Amsterdam": mockOffers,
  "Colone": mockOffers
};

export const reviewsMock = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const reviewItemMock = {
  id: 1,
  user: {
    id: 4,
    isPro: false,
    name: `Max`,
    avatarUrl: `img/1.png`
  },
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
};
