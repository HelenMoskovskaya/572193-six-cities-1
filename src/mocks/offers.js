export default [
  {
    city: `Paris`,
    centerCityCoords: [48.85881005, 2.32003101],
    offerCoords: [48.83832557, 2.29878187],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Paris`,
    centerCityCoords: [48.85881005, 2.32003101],
    offerCoords: [48.86905515, 2.36973166],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Cologne`,
    centerCityCoords: [50.938361, 6.959974],
    offerCoords: [50.92812784, 6.9745481],
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    image: `img/apartment-03.jpg`,
    price: 180,
    rating: 100,
    isPremium: true,
  },
  {
    city: `Cologne`,
    centerCityCoords: [50.938361, 6.959974],
    offerCoords: [50.94851179, 6.9483161],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Brussels`,
    centerCityCoords: [50.8465573, 4.351697],
    offerCoords: [50.82894742, 4.37661409],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Amsterdam`,
    centerCityCoords: [52.38333, 4.9],
    offerCoords: [52.3909553943508, 4.85309666406198],
    title: `Beautiful luxurious apartment at great location`,
    type: `Apartment`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 93,
    isPremium: true,
  },
  {
    city: `Amsterdam`,
    centerCityCoords: [52.38333, 4.9],
    offerCoords: [52.3909553943508, 4.929309666406198],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Hamburg`,
    centerCityCoords: [53.550341, 10.000654],
    offerCoords: [53.55453394, 10.01213372],
    title: `Beautiful luxurious apartment at great location`,
    type: `Apartment`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 93,
    isPremium: true,
  },
  {
    city: `Dusseldorf`,
    centerCityCoords: [51.2254018, 6.7763137],
    offerCoords: [51.22621102, 6.78346485],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Dusseldorf`,
    centerCityCoords: [51.2254018, 6.7763137],
    offerCoords: [51.23314105, 6.77579641],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
];
