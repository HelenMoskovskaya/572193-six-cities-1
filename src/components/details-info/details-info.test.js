import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {DetailsInfo} from './details-info.jsx';
import {mockOffer} from '../../mocks/mocks.js';


const renderer = new ShallowRenderer();

describe(`DetailsInfo`, () => {
  it(`renders correctly`, () => {
    renderer.render(
        <DetailsInfo
          offer={mockOffer}
          changeFavorites={jest.fn()}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
