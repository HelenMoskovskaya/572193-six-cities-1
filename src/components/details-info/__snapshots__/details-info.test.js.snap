// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsInfo renders correctly 1`] = `
<React.Fragment>
  <div
    className="property__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      Waterfront with extraordinary view
    </h1>
    <BookMark
      changeFavorites={[Function]}
      className="property__bookmark-button"
      isFavorite={false}
    />
  </div>
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "60%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      2.7
    </span>
  </div>
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      hotel
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      2
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      8
    </li>
  </ul>
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      393
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Breakfast
      </li>
      <li
        className="property__inside-item"
      >
        Washer
      </li>
      <li
        className="property__inside-item"
      >
        Laptop friendly workspace
      </li>
    </ul>
  </div>
</React.Fragment>
`;
